### ISSUE TITLE
evaluate WebRTC / javascript as backend network stack

### ISSUE THREAD

**Issue by martinjaggi** [epfml/DeAI#12]
as suggested by @Saipraneet, as an alternative to stand-alone applications frameworks such as libp2p, see #3 , we should evaluate browser-based communication frameworks. this could be advantageous in that it might be easier to get it to run on desktop and mobiles.

webRTC is supported in modern browsers. on top of it, there are several javascript [peerJS](https://peerjs.com/), and in particular [simple-peer](https://github.com/feross/simple-peer) looks very suitable.

could this be suitable for communicating gradients as in ML training, supporting desktop or mobile phone OSs (e.g. coreML?), and for interfacing with e.g. PyTorch or similar schemes?

if so, we could try sth like this to build a p2p communication prototype #5 ? what do people think? 
seems we need some help here...

**Comment by martinjaggi** [comment]
interesting that the js version of [libp2p](https://github.com/libp2p/js-libp2p/) can also use WebRTC as a communication backend

**Comment by blagojce95** [comment]
Me and @mmilenkoski will pick up this issue. Also, can you please add me to the organization? :)

**Comment by mmilenkoski** [comment]
[simple-peer](https://github.com/feross/simple-peer) seems like a good and easy way to this. However, it requires a signaling server to exchange signaling data between the two browsers until a peer-to-peer connection is established. Do you think this is acceptable in our case?

[simple-signal](https://github.com/t-mullen/simple-signal) is one possible solution we could use for this purpose.

**Comment by martinjaggi** [comment]
thanks a lot guys! (and i added blagoj to the org as well).

just linking the [peerjs](https://peerjs.com/) pull request here #16

**Comment by martinjaggi** [comment]
i'll close this for now, as our [mobile prototype](https://github.com/epfml/DeAI/tree/master/mobile-browser-based-version) is ready (for now going browser-based with simpler peer.js for communication backend, and TF.js for the neural nets)
